### PR TITLE
Wrap various calls to sigh_command in new macros in tools.oss.bzl

### PR DESCRIPTION
**Pull request by csilvestrini**
sigh isn't imported internally, so we don't want to call it directly. Instead we call these wrappers in tools.oss.bzl, which are easily swappable for new implementations.

### CODE REVIEW

**Comment by csilvestrini** [comment]
CC: @cromwellian @bgogul

**Comment by csilvestrini** [comment]
cl/325121621

File: third_party/java/arcs/build_defs/internal/tools.oss.bzl [hunk 1]
```diff
@@ -15,21 +19,73 @@ def arcs_manifest_parse_test(name, srcs, deps = []):
...
-        progress_message = "Checking Arcs manifest",
-        execute = False,
+        outs = outs,
+        progress_message = "Generating Kotlin Plans",
+        sigh_cmd = sigh_cmd,
+        deps = deps,
+    )
+
+# buildifier: disable=function-docstring
+def arcs_tool_manifest2json(name, srcs, outs, deps):
```
Comment (alxrsngrtn) [inline third_party/java/arcs/build_defs/internal/tools.oss.bzl:52]: I think we can get rid of this rule (in a future PR).

File: third_party/java/arcs/build_defs/internal/tools.oss.bzl [hunk 2]
```diff
@@ -15,21 +19,73 @@ def arcs_manifest_parse_test(name, srcs, deps = []):
...
 
     sigh_command(
         name = name,
         srcs = srcs,
-        deps = deps + [":" + manifest_name],
-        sigh_cmd = "run manifestChecker " + test_args,
-        progress_message = "Checking Arcs manifest",
-        execute = False,
+        outs = outs,
+        progress_message = "Generating Kotlin Plans",
```
Comment (alxrsngrtn) [inline third_party/java/arcs/build_defs/internal/tools.oss.bzl:46]: Optional: Modify the msg to indicate if we're generating Kotlin Plans or protos.

File: third_party/java/arcs/build_defs/internal/tools.oss.bzl [hunk 3]
```diff
@@ -15,21 +19,73 @@ def arcs_manifest_parse_test(name, srcs, deps = []):
         if not src.endswith(".arcs"):
```
Comment (alxrsngrtn) [inline third_party/java/arcs/build_defs/internal/tools.oss.bzl:19]: Optional: In the spirit of minimizing validation logic, shall we remove this? 

-- Actually, this is going to be deleted in a different PR, right?

File: third_party/java/arcs/build_defs/internal/tools.oss.bzl [hunk 4]
```diff
@@ -15,21 +19,73 @@ def arcs_manifest_parse_test(name, srcs, deps = []):
...
+        srcs = srcs,
+        outs = outs,
+        deps = deps,
+        progress_message = "Serializing manifest",
+        sigh_cmd = "manifest2proto --quiet --outdir $(dirname {OUT}) --outfile $(basename {OUT}) {SRC}",
+    )
+
+# buildifier: disable=function-docstring
+def arcs_tool_schema2wasm(name, srcs, outs, deps, language_name, language_flag, wasm, test_harness):
+    # TODO: generated header guard should contain whole workspace-relative
```
Comment (alxrsngrtn) [inline third_party/java/arcs/build_defs/internal/tools.oss.bzl:75]: I see this TODO was moved with the sigh command. 

Is it being tracked anywhere or by anyone?

File: src/tools/tests/BUILD [hunk 5]
```diff
@@ -1,11 +1,11 @@
+load("//third_party/java/arcs/build_defs:arcs_ts_test.bzl", "arcs_ts_test")
 load(
     "//third_party/java/arcs/build_defs:build_defs.bzl",
     "arcs_cc_schema",
     "arcs_kt_gen",
     "arcs_kt_jvm_test_suite",
     "arcs_kt_schema",
     "arcs_proto_plan",
```
Comment (alxrsngrtn) [inline src/tools/tests/BUILD:8]: I think it's somewhat of an overload for folks to look to import thinks outside of `build_defs:build_defs`. Can we import `arcs_ts_test` in `build_defs` to make this streamlined? 

Was this a deliberate decision? If so, what's the reasoning behind it?

File: third_party/java/arcs/build_defs/arcs_ts_test.bzl [hunk 6]
```diff
@@ -0,0 +1,14 @@
+"""Macro for running TypeScript tests with bazel."""
+
+load(":sigh.bzl", "sigh_command")
+
+def arcs_ts_test(name, src, deps, flaky = False):
```
Comment (alxrsngrtn) [inline third_party/java/arcs/build_defs/arcs_ts_test.bzl:5]: Why not add this as another `arcs_tool`? 

Optional: Should we add a `test_only` attribute here, defaulted to `1`?

File: third_party/java/arcs/build_defs/internal/tools.oss.bzl (hunk 1 above)
Comment (csilvestrini) [reply third_party/java/arcs/build_defs/internal/tools.oss.bzl:52]: Good idea, will do

File: third_party/java/arcs/build_defs/internal/tools.oss.bzl (hunk 2 above)
Comment (csilvestrini) [reply third_party/java/arcs/build_defs/internal/tools.oss.bzl:46]: Done

File: third_party/java/arcs/build_defs/internal/tools.oss.bzl (hunk 3 above)
Comment (csilvestrini) [reply third_party/java/arcs/build_defs/internal/tools.oss.bzl:19]: Yeah it's getting deleted

File: third_party/java/arcs/build_defs/internal/tools.oss.bzl (hunk 4 above)
Comment (csilvestrini) [reply third_party/java/arcs/build_defs/internal/tools.oss.bzl:75]: I have no idea what this TODO means, sorry. I just kept it because it was there.

File: src/tools/tests/BUILD (hunk 5 above)
Comment (csilvestrini) [reply src/tools/tests/BUILD:8]: `arcs_ts_test` isn't used internally (and doesn't work). In the import CL I've excluded it from the Copybara config so that it won't appear. That was the reason I took it out of build_defs.bzl

File: third_party/java/arcs/build_defs/arcs_ts_test.bzl (hunk 6 above)
Comment (csilvestrini) [reply third_party/java/arcs/build_defs/arcs_ts_test.bzl:5]: `_test` targets are testonly automatically, I believe. See https://docs.bazel.build/versions/master/be/common-definitions.html

Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/arcs_ts_test.bzl:5]: The linked document refers to things common to all `rules`. Do they mean all `rules` and `macros`?

Comment (alxrsngrtn) [reply third_party/java/arcs/build_defs/arcs_ts_test.bzl:5]: For example, we don't get `deps` or `data` for free in macros, which is why we've added them explicitly. 

https://docs.bazel.build/versions/master/be/common-definitions.html#common-attributes

Comment (csilvestrini) [reply third_party/java/arcs/build_defs/arcs_ts_test.bzl:5]: All BUILD targets ending in `_test` or `-test` are considered `testonly`, regardless of whether they were generated by a rule or a macro.

Here's the error message you get when you try to add a test target created via `arcs_ts_test` as a dep in a non-test target:

```
ERROR: /Users/csilvestrini/dev/arcs/src/tools/tests/BUILD:67:16: in run_in_repo rule //src/tools/tests:variable_proto: non-test target '//src/tools/tests:variable_proto' depends on testonly target '//src/tools/tests:schema-generator-test' and doesn't have testonly attribute set
```
